### PR TITLE
[TF FE] Optimize and correct FusedBatchNorm translator - avoid Transposes

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Optimize and correct FusedBatchNorm translator - avoid Transposes. Add support for `is_training=True` mode.

**Ticket:** 70339

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_FusedBatchNorm.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        dict(x_shape=[2, 3, 4, 5], epsilon=0.0005, exponential_avg_factor=0.3, data_format="NHWC", is_training=True,
+             fbn_version="v2"),
+        dict(x_shape=[3, 2, 1, 5], epsilon=0.00003, exponential_avg_factor=0.7, data_format="NCHW", is_training=True,
+             fbn_version="v3"),
+        dict(x_shape=[3, 4, 2, 5], epsilon=0.0003, exponential_avg_factor=0.7, data_format="NCHW", is_training=True,
+             fbn_version="v3"),
+        # due to limitation in the layer test infrastructure - it finds tensor names for Parameter and Result nodes
+        # by get_any_name() that cannot work if some nodes fused to Parameter or Result node
+        # for inference mode, mean and variance go to outputs as batch_mean and batch_variance
+        # Now this casae will be validated with TF2 Keras BatchNormalization tests
```
Comment (mvafin) [inline tests/layer_tests/tensorflow_tests/test_tf_FusedBatchNorm.py:79]: Maybe mark those tests as skip at least? Not just comment. Also does it mean that `training=false` case is not tested? Is it hard to fix tests to validate tensor `names` not `any_name`?

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_FusedBatchNorm.py:79]: Agree, I will rather to create a bug for me and mark them as xfail.

Comment (rkazants) [reply tests/layer_tests/tensorflow_tests/test_tf_FusedBatchNorm.py]: Fixed.

**Review by mvafin — APPROVED** [review]

File: src/frontends/tensorflow/src/op/fused_batch_norm.cpp [hunk 2]
```diff
@@ -3,55 +3,263 @@
...
+
+    // create fictious output for reserved outputs of FusedBatchNorm operation
+    auto zero_const = make_shared<Constant>(scale.get_element_type(), Shape{}, 0);
+
+    // set node names and tensor names
+    set_node_name(node.get_name(), fused_batch_norm.get_node_shared_ptr());
+    set_node_name(node.get_name() + ":1", batch_mean.get_node_shared_ptr());
+    set_node_name(node.get_name() + ":2", batch_variance.get_node_shared_ptr());
+    set_node_name(node.get_name() + ":3", zero_const);
+    set_node_name(node.get_name() + ":4", zero_const);
```
[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:252]: no we should not, we use output.get_tensor().add_names({out_name});
Moreover, they are reserved and not used in the models.

File: src/frontends/tensorflow/src/op/fused_batch_norm.cpp [hunk 3]
```diff
@@ -3,55 +3,263 @@
...
+
+    OutputVector results = OutputVector{fused_batch_norm, batch_mean, batch_variance, zero_const, zero_const};
     if (is_v3) {
-        // FusedBatchNormV3 has 6 outputs
-        result.push_back(ng_mean);  // reserve_space_3
+        set_node_name(node.get_name() + ":5", zero_const);
```
[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:256]: Replied

File: src/frontends/tensorflow/src/op/fused_batch_norm.cpp [hunk 4]
```diff
@@ -3,55 +3,263 @@
...
+    if (is_nhwc) {
+        return;
+    }
+
+    // in case NCHW format, we need to unsqueeze the normalizing coefficient by lower dimensions
+    // to have the coefficient of shape [C, 1, 1]
+    // generate axes range for unsqueezing the coefficient
+    auto const_one = make_shared<Constant>(element::i32, Shape{}, 1);
+    auto x_rank_minus_one = make_shared<Subtract>(x_rank, const_one);
+    auto axes = make_shared<Range>(const_one, x_rank_minus_one, const_one, element::i32);
```
[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:51]: we can but, Currently it contains less dimensions to unsqueeze - more optimal.

File: src/frontends/tensorflow/src/op/fused_batch_norm.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (itikhono) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:252]: but what about friendly names? https://github.com/openvinotoolkit/openvino/blob/e105b3ea373968b8c699b397fb30cd1c443e35b1/src/frontends/tensorflow/src/utils.cpp#L20

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:252]: But friendly names (operation names) are not used in principal. The tensor names makes sense. Moreover, in real models these reserved nodes are removed during conversion due to no consumers.

[the remark this answers is not in the document]
Comment (itikhono) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:252]: As far as I know, we still use friendly names for legacy conversion to CNNNetwork

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp:252]: TF FE IR is not supported by the legacy API.
If you think it is critical, I can create separate constants. Okay?

[the remark this answers is not in the document]
Comment (rkazants) [reply src/frontends/tensorflow/src/op/fused_batch_norm.cpp]: Fixed.

**Pull request merged**